pkg/pilot/genericpilot/genericpilot.go

```diff
@@ -83,28 +43,44 @@ func (g *GenericPilot) Run() error {
 	// setup healthz handlers
 	g.serveHealthz()
 
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		wait.Until(g.worker, time.Second, g.Options.StopCh)
-	}()
+	ctrlStopCh := make(chan struct{})
+	defer close(ctrlStopCh)
+	go g.controller.Run(ctrlStopCh)
```

Comment:
'g.controller.Run' returns an error, which needs dealing with.

Reply:
And there probably needs to be something to wait for that function to return.

Reply:
Done